xave-finance/halodao-interface-archive
# features/hdf-18

- new HaloHalo abi
- made APY value and price dynamic
- modified useHaloHalo to support new HaloHalo contract version


How to test:
Make sure you have the latest env file. (updated one pass)
Navigate to `/vesting` 
Make sure everything loads properly


For the calculation cc: @ctverceles . You can check the useHaloHalo hook for more details
```
const currentBlockNumber = await providers?.getDefaultProvider().getBlockNumber()
const genesisTimestamp = Number(await halohaloContract?.genesisTimestamp())
const currentTimestamp = await (await providers?.getDefaultProvider().getBlock(currentBlockNumber)).timestamp
// one year in seconds / 31536000
const timeElapsed = currentTimestamp - genesisTimestamp
// 1 comes from the 1:1 ratio before adding rewards
const priceChange = +haloHaloPrice - 1
const APY = timeElapsed * priceChange
```


If ever you need to check the calculations:
```
HaloHalo Total Supply:  173.642857142857142855
Current HALOHALO Price:  3.230769230769230769
Total HALO in the contract:  561.000000000000000003
Genesis Timestamp:  1619549932

```
